### PR TITLE
Fixes to get migrations running on staging

### PR DESCRIPTION
**Pull request by akostibas**
* Update CircleCI's `config.yml` to run the script that runs the migrations container
* Modify the `ecs-run-migrations-container` script to...
  * Pass a `GO_ENV` environment variable into the container telling it which database configuration to use. Right now, due to an incorrect `database.yml` file, this is hardcoded to "prod" :(
  * We needed to escape backslashes in the container URL

### CODE REVIEW

File: bin/ecs-run-migration-container [hunk 1]
```diff
@@ -33,13 +33,22 @@ fi
 container_definition_json=$(aws ecs describe-task-definition --task-definition "$container_definition" --query 'taskDefinition.containerDefinitions[0]')
 
 # create new task definition with the given image
-task_definition_json=$(echo "$container_definition_json" | perl -pe 's/"image":.+/"image": "'"$image"'",/;')
+task_definition_json=$(echo "$container_definition_json" | perl -pe 's/"image":.+/"image": "'"${image//\//\\/}"'",/;')
```
Comment (brainsik) [inline bin/ecs-run-migration-container]: That's really hard to read. Let's change the perl to use something other than `/` as the delimeter. Like, how about `|` (pipe) instead. We almost certainly won't have a `|` in our image name:

```
perl -pe 's|"image":.+|"image": "'"$image"'",|;')
```

File: bin/ecs-run-migration-container [hunk 2]
```diff
@@ -33,13 +33,22 @@ fi
...
 check_arn "$task_definition_arn"
 
+# set environment variables for migrations container
+# see https://docs.aws.amazon.com/cli/latest/reference/ecs/run-task.html#options
+# - we need to set GO_ENV so that the soda migrate command knows which
+#   configuration to use.
+# TODO: Our database.yml is incorrectly identifying the staging database
+#       as 'prod' right now. Once this is fixed, we can have ECS set an
+#       environment variable based on the $environment in this script.
+overrides_json='{"containerOverrides": [{ "name": "app-migrations-'${environment}'", "environment": [{"name": "GO_ENV", "value": "prod"}] }] }'
```
Comment (brainsik) [inline bin/ecs-run-migration-container]: Why are you overriding the name?

File: bin/ecs-run-migration-container (hunk 1 above)
Comment (akostibas) [reply bin/ecs-run-migration-container]: I accept your even fancier perl :)

File: bin/ecs-run-migration-container (hunk 2 above)
Comment (akostibas) [reply bin/ecs-run-migration-container]: I'm not. I originally left this out, but got an error that the `name` field was omitted. Thinking about it more, I think it needs that in order to know *which* container is being overridden.

**Pull request merged**